lib/asciidoctor/document.rb

```diff
@@ -1047,11 +1047,12 @@ def convert opts = {}
     #end
 
     if doctype == 'inline'
-      # QUESTION should we warn if @blocks.size > 0 and the first block is not a paragraph?
-      if (block = @blocks[0]) && block.content_model != :compound
-        output = block.content
-      else
-        output = nil
+      if (block = @blocks[0])
+        if block.content_model == :compound || block.content_model == :empty
+          warn %(asciidoctor: WARNING: no inline candidate; use the inline doctype to convert a single paragragh, verbatim, or raw block)
```

Comment:
What do you think about adding a link on the documentation about how to use document types and/or making a suggestion on how to fix it (ie. remove doctype from options) ?

Maybe we should print this information only when verbose is turned on ?